src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -4488,6 +4488,8 @@ public static void Remove_Invalid()
         [InlineData("Hello", 'l', '!', "He!!o")]
         [InlineData("Hello", 'e', 'e', "Hello")]
         [InlineData("Hello", 'a', 'b', "Hello")]
+        [InlineData("This is a very nice sentence", 'z', 'y', "This is a very nice sentence")]
+        [InlineData("This is a very nice sentence", 'i', 'I', "ThIs Is a very nIce sentence")]
         public static void Replace_Char_Char(string s, char oldChar, char newChar, string expected)
```

Comment:
Are these sufficiently long to test everything we want to test? As long as we're adding tests, should we not add some that are even longer, e.g. to exercise parts that could operate on, say, 128 chars at a time? You can use a MemberData to generate the inputs programmatically.

Reply:
I can't quite think of something that can operate on 128 chars at a time (it would be 16 `char`s maximum because the size of data AVX2 256-bit instructions can handle at once, or 32 if we get AVX-512; but that isn't available in .NET yet). This should be long enough to check the vectorised paths for up to 256-bit vectorised paths

Reply:
> This should be long enough to check the vectorised paths for up to 256-bit vectorised paths.

Seems like there are several branches in your additions that aren't going to get exercised with these new tests, no?

Reply:
I don't think there's several (at least, in regards with string length). I'll add a few more cases to cover more branches.

Reply:
> at least, in regards with string length

e.g. there's a loop of Vector128 over the length.  Do your additions cover the case where you both go through the loop once and don't jump back to iterate again, and also where you go through the loop multiple times?

Reply:
I've just added a few more cases. Still not entirely sure if I've covered enough 😓